test/dynamo/test_misc.py

```diff
@@ -13286,6 +13287,208 @@ class C:
+
+    def test_multiple_datetime_now_in_graph_stable_compile(self):
+        counter = CompileCounter()
+
+        @torch.compile(backend=counter, fullgraph=True)
+        def fn(x):
+            t1 = datetime.datetime.now()
+            t2 = datetime.datetime.now()
+            elapsed = (t2 - t1).total_seconds()
+            z = x * elapsed
+            t3 = datetime.datetime.now()
+            return z + t3.second
+
+        x = torch.randn(6)
+
+        out1 = fn(x)
+        out2 = fn(x)
+
+        self.assertEqual(out1.shape, x.shape)
+        self.assertEqual(out2.shape, x.shape)
+        self.assertEqual(counter.frame_count, 1)
+
+    def test_datetime_now_in_graph_no_recompile_import_as(self):
+        from datetime import datetime as dt
+
+        counter = CompileCounter()
+
+        @torch.compile(backend=counter, fullgraph=True)
+        def fn(x):
+            t = dt.now()
+            return x + t.second
+
+        x = torch.randn(4)
+
+        # multiple runs shouldn't trigger recompile
+        out1 = fn(x)
+        out2 = fn(x)
+
+        self.assertEqual(out1.shape, x.shape)
+        self.assertEqual(out1.dtype, x.dtype)
+        self.assertEqual(out2.shape, x.shape)
+        self.assertEqual(out2.dtype, x.dtype)
+
+        self.assertEqual(counter.frame_count, 1)
+
+    def test_datetime_elapsed_total_seconds_scalar_used(self):
+        counter = CompileCounter()
+
+        @torch.compile(backend=counter, fullgraph=True)
+        def fn(x):
+            start = datetime.datetime.now()
+            y = x + 1
+            end = datetime.datetime.now()
+            elapsed = end - start
+            factor = elapsed.total_seconds()
+            return y * factor
+
+        x = torch.randn(5, 3)
+
+        out1 = fn(x)
+        out2 = fn(x)
+
+        self.assertEqual(out1.shape, x.shape)
+        self.assertEqual(out1.dtype, x.dtype)
+        self.assertEqual(out2.shape, x.shape)
+        self.assertEqual(out2.dtype, x.dtype)
+
+        self.assertEqual(counter.frame_count, 1)
+
+    def test_datetime_timestamp_scalar_used(self):
+        counter = CompileCounter()
+
+        @torch.compile(backend=counter, fullgraph=True)
+        def fn(x):
+            t = datetime.datetime.now()
+            ts = t.timestamp()
+            return x * ts
+
+        x = torch.randn(2, 2)
+
+        out1 = fn(x)
+        out2 = fn(x)
+
+        self.assertEqual(out1.shape, x.shape)
+        self.assertEqual(out1.dtype, x.dtype)
+        self.assertEqual(out2.shape, x.shape)
+        self.assertEqual(out2.dtype, x.dtype)
+
+        self.assertEqual(counter.frame_count, 1)
+
+    def test_datetime_scalar_cant_be_used_in_control_flow(self):
+        counter = CompileCounter()
+
+        @torch.compile(backend=counter)
+        def fn(x):
+            start = datetime.datetime.now()
+            y = x + 1
+            end = datetime.datetime.now()
+            elapsed = end - start
+            s = elapsed.total_seconds()
+            if s > 0.0:
```

Comment:
Is it possible to graph break here and resume instead of skipping the frame? That logic should be here: https://github.com/pytorch/pytorch/blob/1328a02d2eb26ee67346048ee242327ea90d6315/torch/_dynamo/symbolic_convert.py#L648